### PR TITLE
feat: add initial mechanism for omise

### LINKED ISSUE #2859: Add Omise Payment Gateway for Thailand
Omise is important for our Thai event. Please add it.

More: https://www.omise.co/

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #2859 

#### Short description of what this resolves:
 Provides a basic mechanism for dealing with omise payments.
Note : server to frontend redirection is missing

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by uds5501** [comment]
@mrsaicharan1 this is the initial version for now, please let me know of any changes required according to the backend mechanism

**Comment by mrsaicharan1** [comment]
I guess this is good. @uds5501 Let's just test the server PR as soon as I open it. Will notify so that you can pull the changes.

**Comment by mrsaicharan1** [comment]
Also, let's get it reviewed as soon as we are sure both are working.

**Comment by mrsaicharan1** [comment]
@uds5501 The server part for this is complete. Let's test it out as soon as we're done with this!

**Comment by uds5501** [comment]
@mrsaicharan1 Sure!

File: app/controllers/orders/pending.js [hunk 1]
```diff
@@ -13,10 +13,23 @@ export default Controller.extend({
...
     return this.get('model.order.amount') * 100;
   }),
 
+  publicKeyOmise: computed('settings.omiseLivePublic', function() {
+    return this.get('settings.omiseLivePublic') || this.get('settings.omiseTestPublic');
+  }),
+
+  omiseFormAction: computed('model.order', function() {
+    let identifier = this.get('model.order.identifier');
+    return `http://127.0.0.1:5000/v1/orders/'${identifier}'/omise-checkout`;
```
Comment (mrsaicharan1) [inline app/controllers/orders/pending.js]: `http://127.0.0.1:5000/v1/orders/'${identifier}'/omise-checkout` replace this with `${hostname}/${identifier}/omise-checkout` . You can find hostname from `environment.js`

Comment (uds5501) [reply app/controllers/orders/pending.js]: @mrsaicharan1 Sure! pushing the changes asap

**Comment by uds5501** [comment]
@mrsaicharan1 @CosmicCoder96 ready for review

File: app/controllers/orders/pending.js [hunk 2]
```diff
@@ -15,10 +16,23 @@ export default Controller.extend({
...
+
   paymentAmount: computed('model.order', function() {
     return this.get('model.order.amount') * 100;
   }),
 
+  publicKeyOmise: computed('settings.omiseLivePublic', function() {
+    return this.get('settings.omiseLivePublic') || this.get('settings.omiseTestPublic');
+  }),
+
+  omiseFormAction: computed('model.order', function() {
```
Comment (niranjan94) [inline app/controllers/orders/pending.js]: Be specific when watching other properties. If you're using `model.order.identifier` then put that in the computed property.

File: app/controllers/orders/pending.js [hunk 3]
```diff
@@ -15,10 +16,23 @@ export default Controller.extend({
...
 
+  isOmise: computed('model.order', function() {
+    return this.get('model.order.paymentMode') === 'omise';
+  }),
+
   paymentAmount: computed('model.order', function() {
     return this.get('model.order.amount') * 100;
   }),
 
+  publicKeyOmise: computed('settings.omiseLivePublic', function() {
```
Comment (niranjan94) [inline app/controllers/orders/pending.js]: Add `settings.omiseTestPublic`

File: app/controllers/orders/pending.js [hunk 4]
```diff
@@ -15,10 +16,23 @@ export default Controller.extend({
     return this.get('model.order.paymentMode') === 'paypal';
   }),
 
+  isOmise: computed('model.order', function() {
```
Comment (niranjan94) [inline app/controllers/orders/pending.js]: Be specific. Use `model.order.paymentMode`

**Comment by uds5501** [comment]
@niranjan94 thanks for the review! I have specified the watched properties. Please take a look!

**Comment by mrsaicharan1** [comment]
@uds5501 Fix up travis and we can merge this.

**Comment by uds5501** [comment]
@mrsaicharan1 It's that random travis fail, idk how to handle this.

**Pull request merged**